Tribler/community/tunnel/pooled_tunnel_community.py

```diff
@@ -0,0 +1,345 @@
+        :return:
+        """
+        self.trsession = tribler_session
+        self.settings = settings if settings\
+            else TunnelSettings(tribler_session=tribler_session)
+
+        super(PooledTunnelCommunity, self).initialize()
+
+        self.socks_server =\
+            Socks5Server(self,
+                         tribler_session.get_tunnel_community_socks5_listen_ports()
+                         if tribler_session else self.settings.socks_listen_ports)
+        self.socks_server.start()
+
+        if self.trsession:
+            self.notifier = self.trsession.notifier
+
+        # Multiply the single TunnelSettings by the amount of
+        # workers we can handle.
+        self.pool = ProcessManager(tribler_session, self)
+        suggested_workers = self.pool.get_suggested_workers()
+        self.pool.set_worker_count(suggested_workers)
+        self.settings.min_circuits *= suggested_workers
+        self.settings.max_circuits *= suggested_workers
+
+        self.register_task("do_circuits",
+                           LoopingCall(self.do_circuits)).start(5, now=True)
+
+    def unload_community(self):
+        """Callback for when this community should be unloaded
+
+        :returns: None
+        """
+        self.pool.set_worker_count(0)
+        super(PooledTunnelCommunity, self).unload_community()
+
+    @classmethod
+    def get_master_members(cls, dispersy):
+        """Return the normal HiddenTunnelCommunity master member
+
+        :param dispersy: the dispersy instance to use
+        :type dispersy: Tribler.dispersy.dispersy.Dispersy
+        :return: the master member
+        :rtype: Tribler.dispersy.member.Member
+        """
+        return HiddenTunnelCommunity.get_master_members(dispersy)
+
+    def initiate_conversions(self):
+        """Create the conversions for this community
+
+        Since it doesn't communicate, we need none.
+
+        :return: the list of our conversions
+        :rtype: [Tribler.dispersy.conversion.Conversion]
+        """
+        return [DefaultConversion(self)]
+
+    @property
+    def dispersy_enable_candidate_walker(self):
+        """Disable sending introduction-requests
+
+        :return: whether we should enable introduction-requests
+        :rtype: bool
+        """
+        return False
+
+    @property
+    def dispersy_enable_candidate_walker_responses(self):
+        """Disable sending introduction-responses
+
+            :return: whether we should enable introduction-responses
+            :rtype: bool
+        """
+        return False
+
+    def init_circuit(self, circuit_id, circuit):
+        """Callback for when a synced circuit is localized
+
+        :param circuit_id: the circuit's id
+        :type circuit_id: long
+        :param circuit: the circuit
+        :type circuit: Tribler.community.tunnel.remotes.circuit.Circuit
+        :returns: None
+        """
+        setattr(circuit, 'proxy', self)
```

Comment:
This is an initializer of an object created with `__new__()` (bypassing the `__init__()` to provide a snapshot of the remote object), the `.proxy` field does not exist (yet), but needs to be localized on the current process.